Framework/Algorithms/src/DiscusMultipleScatteringCorrection.cpp

```diff
@@ -1807,11 +1835,13 @@ DiscusMultipleScatteringCorrection::createSparseWorkspace(const API::MatrixWorks
 void DiscusMultipleScatteringCorrection::createInvPOfQWorkspaces(ComponentWorkspaceMappings &matWSs, size_t nhists) {
   for (auto &SQWSMapping : matWSs) {
     auto &QSQ = SQWSMapping.QSQ;
-    size_t expectedMaxSize = QSQ->size();
-    MatrixWorkspace_sptr ws = DataObjects::create<Workspace2D>(nhists, HistogramData::Points{0.});
-    ws->dataX(0).reserve(expectedMaxSize);
+    size_t expectedMaxSize =
+        std::accumulate(QSQ->histograms().begin(), QSQ->histograms().end(), static_cast<size_t>(0),
```

Comment:
```suggestion
        std::accumulate(QSQ->histograms().cbegin(), QSQ->histograms().cend(), static_cast<size_t>(0),
```

Reply:
Yes will change. I copied the code from Workspace2D::size() where it uses the non-const iterators for some reason